File: 2020/1/main.py
from aoc import input

def solve(numbers: list[int]) -> int | None:
  for i in range(len(numbers)):
    for j in range(len(numbers)):
      if i != j and numbers[i] + numbers[j] == 2020:
        return numbers[i] * numbers[j]

def main():
  print(solve(input.ints))

if __name__ == "__main__":
  main()

File: 2020/1/main.go
package main

import (
	"fmt"

	"github.com/alexg-axis/advent-of-code/utils/go/parsing"
)

func main() {
	input, err := parsing.ReadInput()
	if err != nil {
		panic(err)
	}

	numbers, err := input.Ints()
	if err != nil {
		panic(err)
	}

	fmt.Println(solve(numbers))
}

func solve(numbers []int) int {
	for i, a := range numbers {
		for j, b := range numbers {
			if i != j && a+b == 2020 {
				return a * b
			}
		}
	}
	return -1
}
